certora/specs/RevertsERC20.spec

```diff
@@ -8,40 +8,31 @@ methods {
     function _._moveDelegateVotes(address, address, uint256) internal => CONSTANT;
 }
 
-// Check that transfers with zero address or too large amounts revert.
-rule transferReverts(env e, address to,uint256 amount) {
-    // Safe require as transfer is a non-payable function.
-    require e.msg.value == 0;
-
+// Check the revert conditions for the transfer function.
+rule transferRevertConditions(env e, address to,uint256 amount) {
     uint256 balanceOfSenderBefore = balanceOf(e.msg.sender);
 
     transfer@withrevert(e, to, amount);
-    assert !lastReverted <=> e.msg.sender != 0 && to !=0 && balanceOfSenderBefore >= amount;
+    assert lastReverted <=> e.msg.sender == 0 || to == 0 || balanceOfSenderBefore < amount || e.msg.value != 0;
 }
 
-
-// Check that transfersFrom with zero address or too large allowances revert.
-rule transferFromReverts(env e, address from, address to,uint256 amount) {
-    // Safe require as transferFrom is a non-payable function.
-    require e.msg.value == 0;
-
+// Check the revert conditions for the transferFrom function.
+rule transferFromRevertConditions(env e, address from, address to,uint256 amount) {
```

Comment:
```suggestion
rule transferFromRevertConditions(env e, address from, address to, uint256 amount) {
```

Reply:
Fixed in https://github.com/morpho-org/morpho-token-upgradeable/pull/98/commits/9025f3b5c34e16082305e6d4b97dc5bf96eb67f5.